Taskfile.yml

```diff
@@ -111,54 +113,70 @@ tasks:
     method: timestamp
     cmds:
       - "{{.BOOT_PART_SCRIPT}} --kernel {{.KERNEL_FILE}} -o {{.BOOT_PART}}"
-    status:
-      - "{{.CONFIG_HELPER}} boot-part {{.BOOT_PART_SCRIPT}}"
-      - "test -f {{.BOOT_PART}}"
-    preconditions:
-      - *check-config
     run: once
     vars:
       BOOT_PART_SCRIPT: ./scripts/boot_partition.sh
 
   data-part:
+    deps:
+      - check-config
     sources:
+      - "{{.CONFIG}}"
       - "{{.ST_LOCAL_OSPKG_DIR}}/**"
     generates:
       - "{{.DATA_PART}}"
     method: timestamp
     cmds:
       - "{{.DATA_PART_SCRIPT}} -o {{.DATA_PART}}"
     status:
-      - "{{.CONFIG_HELPER}} data-part {{.DATA_PART_SCRIPT}}"
+      - "{{.CONFIG_HELPER}} data-part {{.DEPS}}"
+      - "[ -f {{.DATA_PART}} ] && [ -z \"$(ls -A {{.ST_LOCAL_OSPKG_DIR}})\" ]"
     preconditions:
-      - *check-config
       - sh: '[ -z "$ST_BOOT_MODE" ] || [ "$ST_BOOT_MODE" != "local" ] || (find {{.ST_LOCAL_OSPKG_DIR}} -name \*.zip | read)'
-        msg: "[ERROR] Path \"{{.ST_LOCAL_OSPKG_DIR}}\" does not contain any OS package, which is required in Local Boot Mode.\n
-        Please provide an OS Package or run \"task demo:ospkg\" to generate an example OS Package"
+        msg: |
+          [ERROR] Path "{{.ST_LOCAL_OSPKG_DIR}}" does not contain any OS package, which is required in Local Boot Mode.
+          Please provide an OS Package or run "task demo:ospkg" to generate an example OS Package"
```

Comment:
I think you forgot to remove the last `"`

Reply:
ack